### PR TITLE
feature/GH-13403: Move OrderOverviewComponent from shared to order folder

### LINKED ISSUE #13403: Move OrderOverviewComponent from shared to order folder
`OrderOverviewComponent` is in `shared` folder. We will move it inside `order` folder first, then move it to `order` lib. And make `checkout` lib have cms mapping map from ConfirmationOrderOverview to OrderOverviewComponent.

### PR DESCRIPTION
**Pull request by WeizhengSap**
close #13403

### CODE REVIEW

File: feature-libs/checkout/components/order-confirmation/order-confirmation.module.ts [hunk 1]
```diff
@@ -60,7 +59,13 @@ const orderConfirmationComponents = [
           guards: [OrderConfirmationGuard],
         },
         OrderConfirmationOverviewComponent: {
-          component: OrderConfirmationOverviewComponent,
+          component: OrderDetailShippingComponent,
```
Comment (WeizhengSap) [inline feature-libs/checkout/components/order-confirmation/order-confirmation.module.ts:62]: Is it breaking change?

Comment (plabadie) [reply feature-libs/checkout/components/order-confirmation/order-confirmation.module.ts:62]: I'm affraid this looks like a breaking change to me. 
We can discuss.

File: projects/storefrontlib/src/cms-components/myaccount/order/order-details/order-overview/order-overview.module.ts [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+
+/**
+ * @deprecated since 4.1 - use order lib instead
+ */
+@NgModule({
+  imports: [CommonModule, I18nModule, CardModule],
+  declarations: [OrderOverviewComponent],
+  exports: [OrderOverviewComponent],
+})
+export class OrderOverviewModule {}
```
Comment (plabadie) [inline projects/storefrontlib/src/cms-components/myaccount/order/order-details/order-overview/order-overview.module.ts:15]: General comment for the PR:
There was [a plan](https://github.com/SAP/spartacus/issues/13324) to wait until 5.0 and do the following:
- move `OrderOverviewComponent` to the order-lib
- change the checkout-lib to depend on order-lib

Also, we agreed to move the `OrderOverviewComponent` out of order-lib and back into storefront lib in [a recent PR](https://github.com/SAP/spartacus/pull/13363/files). Why move it back to the lib? 

Lets discuss.

File: feature-libs/checkout/components/order-confirmation/components/order-confirmation-overview/order-confirmation-overview.component.ts [hunk 3]
```diff
@@ -7,6 +7,9 @@ import {
 import { CheckoutFacade } from '@spartacus/checkout/root';
 import { Observable } from 'rxjs';
 
+/**
+ * @deprecated since 4.1 - This component is not used any more, but use `OrderDetailShippingComponent` instead
+ */
```
Comment (plabadie) [inline feature-libs/checkout/components/order-confirmation/components/order-confirmation-overview/order-confirmation-overview.component.ts:12]: We need to make these comments 4.2 now.

**Comment by plabadie** [comment]
Closing.  We agreed to postpose such changes to the next major version.

**Pull request closed without merging**